### ISSUE TITLE
Treat busways mostly like roads

### ISSUE THREAD

**Issue by Aeris1One** [streetcomplete/StreetComplete#5749]
**Use case**
Busways (`highway=busway`) aren't rendered and aren't considered like roads. There aren't quests on them (while there may be crossroads, bus stops, bike lanes...). They should be treated just like normal roads for most of the quests, except some like number of lanes, speed limit...

You can look at Rouen, France to see a busway in urban area (https://www.openstreetmap.org/#map=17/49.43990/1.09030&layers=P, do not change the layers, default OSM style doesn't render busways)
Rouen's busway is also allowed to bikes, a quest for busways asking if bicycles, taxis and/or deliveries are allowed could be added.

Here's a zoom onto a part of Rouen's main busway :
![ResizedImage_2024-07-17_21-02-16_3667](https://github.com/user-attachments/assets/d3e19f38-6247-4c34-b7bf-1c12aa61d140)
*There is a busway in the blank, and a bus station below the building on which there's a clock quest*

**Proposed Solution**
- render busways on the map
- enable a subset of the main quests on such ways (most quests except some like number of lanes or speed limit)
- (eventually) add a quest asking if bicycles/taxis/deliveries are allowed

**Comment by matkoniecz** [comment]
I guess that adding busways to address quest would be a good first step... See https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddRoadSurface.kt

> enable a subset of the main quests on such ways (most quests except some like number of lanes or speed limit)

which ones except surface quest?

> render busways on the map

are these in Jawg vector tiles?

> (eventually) add a quest asking if bicycles/taxis/deliveries are allowed

opening a new issue for quests and checking whether requirements are fulfilled would be a first step here

**Comment by mnalis** [comment]
>> - render busways on the map

> are these in Jawg vector tiles?

BTW one can check Jawg vector tiles at https://streetcomplete.app/map-jawg/ to verify, but by my quick look at e.g. [w344099422](https://www.openstreetmap.org/way/344099422) it does not seems to render `highway=busway`. I've tried [playing](https://streetcomplete.github.io/streetcomplete-mapstyle/), but (unless I'm wrong) it does not seem [JawgMaps](https://www.jawg.io) includes it in vector tiles regardless of styling, so one would  have to ask them if they would add support for it.

> opening a new issue for quests and checking whether requirements are fulfilled would be a first step here

As a note there, I'm (personally) afraid that asking quests on things which are **_invisible on the map_** could be quite confusing for average StreetComplete user; so getting Jawg to render `highway=busway` might likely be a prerequisite for adding/modifying those quests too.

**Comment by Aeris1One** [comment]
> As a note there, I'm (personally) afraid that asking quests on things which are invisible on the map could be quite confusing for average StreetComplete user

Agreed

> which [quests] except surface quest?
There might be bus stops on such ways, so quests like "is there a bench", "is there a bin", etc.
There might also be crossroads (so "is it painted", "is it wheelchair-accessible", etc.)
They may be allowed to bicycles, or forbidden. They might be allowed to pedestrians, or not. Also, taxis (that might be a quest specific to busway though).

In fact I do think most quests that apply to normal roads could be enabled on busway as well. There are only a few that do not apply to busways : speedlimits as those busway might use a different signalling system (like trams) that StreetComplete users aren't necessarily able to read, and maybe number of lanes (can't think of any busway that have more than 2 lanes).

**Comment by matkoniecz** [comment]
> There might be bus stops on such ways, so quests like "is there a bench", "is there a bin", etc.

these are asked for bus stops, not roads with buses

>  There might also be crossroads (so "is it painted", "is it wheelchair-accessible", etc.)

asked for crossings, including crossing on busways IIRC

>  They may be allowed to bicycles, or forbidden. They might be allowed to pedestrians, or not. Also, taxis (that might be a quest specific to busway though).

opening a new issue for quests and checking whether requirements are fulfilled would be a first step here

**Comment by matkoniecz** [comment]
> >  As a note there, I'm (personally) afraid that asking quests on things which are invisible on the map could be quite confusing for average StreetComplete user

> Agreed

then I will close it, pending https://github.com/streetcomplete/streetcomplete-mapstyle/issues/147 (I added note there that at least surface quest could make sense - I guess that also support for `lit` would make sense,,,)

**Comment by mnalis** [comment]
> In fact I do think most quests that apply to normal roads could be enabled on busway as well

Yes, if Jawg implements `highway=busway`, most (but not all; e.g. probably `StreetParkingOverlay.kt` does not make sense on busways) of the quests/overlay using `ALL_ROADS` might be enabled there too (e.g. sidewalks, cycleways, surface, lit, road name, AddBarrierOnRoad).

And there are also some that use their specific filters (instead of `ALL_ROADS`) which might (or should _not_) be implemented: AddRoadSurface, AddShoulder, AddLanes, AddCrossing, AddCycleway, AddSidewalk, AddRoadWidth, AddMaxHeight, AddMaxWeight)

**Comment by riQQ** [comment]
Reopening as Jawg integrated busways in their tiles (see https://github.com/streetcomplete/streetcomplete-mapstyle/issues/147#issuecomment-2247321821)

**Comment by matkoniecz** [comment]
Does it make sense to ask about it in sidewalk and cycleway quests? In my experience busways do not really have those

**Comment by westnordost** [comment]
- `AddRoadName` OK. 62% of busways already have a name
- `AddLanes` OK. Some will have 2, some (the oneway ones) will have 1.
- `AddOneway` OK. 70% of busways are tagged with oneway=yes
- `AddMaxSpeed`, `AddMaxWeight`, `AddMaxHeight` OK. Signage for buses is the same as signage for car traffic in general
- `AddRoadSurface`, `SurfaceOverlay` OK
- `AddRoadSmoothness` OK.
- `AddRoadWidth` Hmm, OK. It is only asked under very specific circumstances anyway, i.e. `lanes < 2` or `lane_markings=no` and also` access != no` (like many others)
- `AddWayLit`, `WayLitOverlay` OK, could make sense for the big picture
- ~`AddSidewalk`~, ~`AddCycleway`~ While theoretically possible, it has zero usage. Oftentimes busways are layouted in the same way trams would be layouted, i.e. in the middle of the road, between the car lanes. Of course there is no sidewalk or cycleway. However, it **should** be asked in the case that the busway has invalid sidewalk or cycleway tagging
- `AddSidewalkSurface` OK. This is only asked for ways that DO have a sidewalk already tagged
- ~`AddProhibitedForPedestrians`~ No need to ask.
- ~`StreetParkingOverlay`~ Can't park there anyway
- `AddCrossing` OK. Pedestrian crossings with busways are possible.
- `AddBarrierOnRoad` OK. If a fence intersects with a busway, it is fair to ask about the type of opening in that barrier for the busway
- `AddKerbHeight`, `AddTactilePavingKerb` OK. (when footway crosses busway)
- `SidewalkOverlay`, ̀ CyclewayOverlay` OK. Theoretically possible after all. But it should not be highlighted in red (i.e. as missing).

**Comment by westnordost** [comment]
See [PR ](https://github.com/streetcomplete/StreetComplete/pull/5797), review welcome.

**Comment by mnalis** [comment]
**TL;DR:** I think it would make sense to ask on `AddSidewalk` / `AddCycleway` quests on `highway=busway` too.

> <Strike>AddSidewalk</strike>, <strike>AddCycleway</strike> While theoretically possible, it has zero usage. 

Somewhat low-ish, but not zero - there are [354 cycleways](https://overpass-turbo.eu/s/1POl) + [856 sidewalks](https://overpass-turbo.eu/s/1POn) already tagged combinations on `highway=busway`, but it is below _minumum-1000-elements-per-tag-to-show_ taginfo requirement, so it doesn't show there (but it does at e.g. overpass).
 
But then again there are only 23k `highway=busway` at the moment, so those 1.2k sidewalk/cycleways are over `5%` already tagged combinations...

> Oftentimes busways are layouted in the same way trams would be layouted, i.e. in the middle of the road, between the car lanes. Of course there is no sidewalk or cycleway. 

In those cases however it probably would not tagged with `highway=busway` (but instead with e.g. `highway=primary` + [busway:right=lane](https://wiki.openstreetmap.org/wiki/Key:busway) or [bus:lanes=no|designated|no](https://wiki.openstreetmap.org/wiki/Key%3Abus%3Alanes) or similar) so the issue does not arise (i.e. it would still be asked by regular `highway=primary` AddSidewalk/AddCycleway rules)
